### DISCUSSION TITLE
Semester 4 Week 3

### DISCUSSION THREAD

**Discussion by Stev3eM** [SE-TINF22B6/CookHub#159] [Show and tell]
Hello backpackers,

This is our weekly discussion-blog to keep you up to date within our project progression.
A new week has approched and we got again some new updates for you:

<br><br>

## Weekly progress

<br>

### Backend / API 
✅ Stored the API key into a config file to be ignored by git to avoid exposing sensitive data
✅ Improved recipe API search function
✅ Added API endpoint for the returning of the top n recipes (e.g. recommendations)

### MyRecipe Page
✅ Redesign of the page to improve user experience and usage
✅ Replaced slider with a radio button group to select recipe difficulty
✅ Replaced star rating with a toggleable heart-button
✅ Implemented a table to show different adventurerized versions of a recipe

### Bug Fixes
✅  Fixed several Bugs like:
- Different sized elements on pages led to strange structure on different screen sizes
- Adjustments on several css files

<br><br>

## Project Structure

<br>

> [!note] 
> In addition we provide you information about the project structure:

<br>

> _We decided to use a layered architecture._
>
> The layered architecture is a software design pattern that organizes applications into separate layers, each responsible for distinct aspects of the application's functionality. This architecture is renowned for its simplicity, low cost, and reliability, making it ideal for our project.
>
> _Why we have chosen this architecture?_ 🦖
> 
> - Separation of Concerns: Simplifies complexity by dividing responsibilities across layers.
> - Modularity: Facilitates independent development and maintenance of each layer.
> - Simplicity: It is easy to set up and ensures rapid development.
> - Enhanced Security: Security measures can be enforced at different levels within the architecture.
> - Development Flexibility: Parallel development becomes feasible with minimal dependencies across teams.
> - Simplified Deployment: Updates to one layer can often be made without affecting others.

<br>

#### Sequence diagram of the OpenAI API access:
![image](https://github.com/SE-TINF22B6/CookHub/raw/main/Project_Doc/Diagrams/openai-sequence-diagram.drawio.svg)

#### Package diagram
![image](https://raw.githubusercontent.com/SE-TINF22B6/CookHub/6359bde867a9dcb0d8e79b7a5503971673a0f6c6/Project_Doc/Diagrams/package-diagram.drawio.svg)

#### Component diagram
![image](https://raw.githubusercontent.com/SE-TINF22B6/CookHub/efa0b47abc95072286463ddcf41b16e0a9d15486/Project_Doc/Diagrams/component_diagram.drawio.svg)

<br><br>

See you next time!

**Your**   <img width="50" alt="image" src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/1a218c37-d3b0-4c88-9ae1-827b5a4eba42">   **Team**

**Comment by TomSarh** [comment]
Dear CookHub team,

This weekly progress report is a testament to your commitment to continuously improve CookHub ! The user experience improvements are always welcome and securing the API is a great touch too. Other than that, great job and keep it up !

**Comment by MargotGsb** [comment]
Dear Team CookHub,

Your progress this week is amazing. You developed a lot of things and additionally your diagramms look very good and explain your project structure very well. I think you made a good decision with the layered architcture. Keep this work up!

Your Team Plapy :D

**Comment by HowDoIUseGit** [comment]
Moin CookHub,

Great work this week! You've made significant improvements across the board. From securing sensitive data in the backend to enhancing user experience on the MyRecipe page and fixing various bugs, your attention to detail is evident.
Keep up the great work!

Your CookPit
